core/src/main/java/ai/timefold/solver/core/preview/api/domain/variable/declarative/ShadowSources.java

```diff
@@ -95,4 +95,22 @@
      * @return A non-empty list of variables the supplier method accesses.
      */
     String[] value();
+
+    /**
+     * If non-empty, this is the property name of a problem fact on the entity that will be used
+     * to identify the entity's group.
+     * <p>
+     * When the group key is non-null, the shadow variable will only be calculated
+     * for one entity with that group key, and all other entities with that group key will
+     * have their shadows set to that value.
```

Comment:
This is awkwardly phrased, IMO.
I'd start by saying that, if the value is null, the entities are considered independent and their values will not be aligned. Then continue with the rest.